model/project_ref.go

```diff
@@ -1255,6 +1304,180 @@ func (projectRef *ProjectRef) Upsert() error {
+		}
+	case ProjectRefGithubSection:
+		err = db.Update(ProjectRefCollection,
+			bson.M{ProjectRefIdKey: p.Id},
+			bson.M{
+				"$unset": bson.M{
+					projectRefPRTestingEnabledKey:      1,
+					projectRefGithubChecksEnabledKey:   1,
+					projectRefGithubTriggerAliasesKey:  1,
+					projectRefGitTagVersionsEnabledKey: 1,
+					ProjectRefGitTagAuthorizedUsersKey: 1,
+					ProjectRefGitTagAuthorizedTeamsKey: 1,
+					projectRefCommitQueueKey:           1,
+				},
+			})
+		if err != nil {
+			return errors.Wrapf(err, "error defaulting to repo for section '%s'", section)
+		}
+		catcher := grip.NewBasicCatcher()
+		for _, a := range before.Aliases {
+			// remove only internal aliases
+			if utility.StringSliceContains(evergreen.InternalAliases, a.Alias) {
+				catcher.Add(RemoveProjectAlias(a.ID.Hex()))
+			}
+		}
+		if catcher.HasErrors() {
+			// log the changes before returning whatever errors were caught
+			_ = getAndLogProjectModified(p.Id, caller, before)
+			return errors.Wrapf(catcher.Resolve(), "error defaulting to repo for section '%s'", section)
+		}
+
+	case ProjectRefNotificationsSection:
+		err = db.Update(ProjectRefCollection,
+			bson.M{ProjectRefIdKey: p.Id},
+			bson.M{
+				"$unset": bson.M{projectRefNotifyOnFailureKey: 1},
+			})
+		if err != nil {
+			return errors.Wrapf(err, "error defaulting to repo for section '%s'", section)
+		}
+		// handle subscriptions
+		catcher := grip.NewBasicCatcher()
+		for _, sub := range before.Subscriptions {
+			catcher.Add(event.RemoveSubscription(sub.ID))
+		}
+		if catcher.HasErrors() {
+			// log the changes before returning whatever errors were caught
+			_ = getAndLogProjectModified(p.Id, caller, before)
+			return errors.Wrapf(catcher.Resolve(), "error defaulting to repo for section '%s'", section)
+		}
+	case ProjectRefAliasSection:
+		catcher := grip.NewBasicCatcher()
+		// remove only non-internal aliases
+		for _, a := range before.Aliases {
+			if !utility.StringSliceContains(evergreen.InternalAliases, a.Alias) {
+				catcher.Add(RemoveProjectAlias(a.ID.Hex()))
+			}
+		}
+		if catcher.HasErrors() {
+			// log any changes before returning whatever errors were caught
+			_ = getAndLogProjectModified(p.Id, caller, before)
+			return errors.Wrapf(catcher.Resolve(), "error defaulting to repo for section '%s'", section)
+		}
+	case ProjectRefTriggersSection:
+		err = db.Update(ProjectRefCollection,
+			bson.M{ProjectRefIdKey: p.Id},
+			bson.M{
+				"$unset": bson.M{
+					projectRefTriggersKey:            1,
+					projectRefPatchTriggerAliasesKey: 1, // is this where we decided patch trigger aliases would go?
+				},
+			})
+		if err != nil {
+			return errors.Wrapf(err, "error defaulting to repo for section '%s'", section)
+		}
+	case ProjectRefWorkstationsSection:
+		err = db.Update(ProjectRefCollection,
+			bson.M{ProjectRefIdKey: p.Id},
+			bson.M{
+				"$unset": bson.M{projectRefWorkstationConfigKey: 1},
```

Comment:
Similar comment as above, we'd need to separate this into WorkstationSetupCommand.Command and WorkstationSetupCommand.Directory if we refactor to one update function.